### DISCUSSION TITLE
Same Identity Multiple Auth Providers

### DISCUSSION THREAD

**Discussion by arnavchaudhary** [fief-dev/fief#217] [Ideas]
Hi,
I was wondering if this is a unique case for me or are other also facing similar issues.

I want to show multiple ways to login eg. Google, Facebook and Username / Password.
If the user with the same email tries accessing the identity should be the same.

Currently, when we try this, it shows the error the User with same email exists already

**Comment by fief-bailiff** [comment]
Hail, @arnavchaudhary 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Hello 👋 

For security reasons, it's considered a best practice to not automatically link existing accounts if the user authenticates with a new social oauth provider. 

In the future, we plan to add the ability for an authenticated user to link an OAuth Provider, directly from its user dashboard.

**Comment by D3visionNL** [comment]
Hi, I was wondering if this is still on the list? I am working on a platform which requires a specific oauth connection and would really like to use Fief as I have found that most other solutions don't provide the same level of control/customizability.

**Reply by frankie567** [reply]
It is! Just not in the works for now 😊
